cmake/Googletest.cmake

```diff
@@ -53,20 +53,26 @@ macro(fetch_googletest)
         set(GTEST_ROOT ${GOOGLETEST_BUILD_ROOT}/googletest-install CACHE path "GoogleTest installation root")
     endif()
 
+    # FindGTest should get call after GTEST_ROOT is set
+    find_package(GTest QUIET)
+
     # https://gitlab.kitware.com/cmake/cmake/issues/17799
-    # FindGtest is buggy when dealing with Debug build. 
+    # FindGtest is buggy when dealing with Debug build.
     if (CMAKE_BUILD_TYPE MATCHES Debug AND GTEST_FOUND MATCHES FALSE)
         message("Setting GTest libraries with debug...")
 
         if (GTEST_LIBRARY_DEBUG MATCHES GTEST_LIBRARY_DEBUG-NOTFOUND)
-                set(gtest_library "")
-                set(gtest_main_library "")
-            if(WIN32)
-                set(gtest_library lib/gtestd.lib)
-                set(gtest_main_library lib/gtest_maind.lib)
-            else()
-                set(gtest_library lib64/libgtestd.a)
-                set(gtest_main_library lib64/libgtest_maind.a)
+            set(gtest_library "")
+            set(gtest_main_library "")
+            if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+                set(gtest_library bin/gtestd.dll)
+                set(gtest_main_library bin/gtest_maind.dll)
+            elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+                set(gtest_library lib/libgtestd.dylib)
+                set(gtest_main_library lib/libgtest_maind.dylib)
+            elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+                set(gtest_library lib64/libgtestd.so)
+                set(gtest_main_library lib64/libgtest_maind.so)
             endif()
                 set(GTEST_INCLUDE_DIRS ${GOOGLETEST_BUILD_ROOT}/googletest-install/include)
```

Comment:
> 
> 
> Shouldn't we be using GTEST_ROOT here, instead of hardcoding `${GOOGLETEST_BUILD_ROOT}/googletest-install`?

Good point! See 26e3834 for fix.

> 
> 
> Also, would it be worth doing a sanity check here to ensure that GTEST_INCLUDE_DIRS / GTEST_LIBRARY_DEBUG / GTEST_MAIN_LIBRARY_DEBUG actually exist? (I mean, they SHOULD, but you never know...)

I see your point but these variables do really get set in Debug and honestly the sanity check is a bit of an overkill.  If you still disagree with me, I can add the checks. 
Honestly, none of these codes were necessary if only FindGTest module that comes with CMake really worked out of the box.

Reply:
Ok, if you think it's overkill, I'm fine with not adding the test.

I see that you changed to GTEST_ROOT, thanks!

One last thing though, since you'd made a lot of similar fixes in this PR - I just noticed the indentation for these lines seems a little wrong.  Shouldn't these 3 lines be de-dented left one level?

Reply:
What are the 3 lines that needs to de-dented?

Reply:
The 3 starting here: https://github.com/Autodesk/maya-usd/blob/b11cb1dec8e635287ceb0e3bfe4f83faa90336d3/cmake/Googletest.cmake#L73

The endif() closes the if-block, but they still have the same indentation as the statements inside the if-block.

Reply:
Done!

Reply:
Fast work!